web/middlewares/secure.go

```diff
@@ -0,0 +1,112 @@
+package middlewares
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cozy/cozy-stack/pkg/instance"
+	"github.com/labstack/echo"
+)
+
+type (
+	XFrameOption string
+	CSPSource    int
+
+	// SecureConfig defines the config for Secure middleware.
+	SecureConfig struct {
+		HSTSMaxAge    time.Duration
+		CSPScriptSrc  []CSPSource
+		CSPFrameSrc   []CSPSource
+		CSPConnectSrc []CSPSource
+		XFrameOptions XFrameOption
+		XFrameAllowed string
+	}
+)
+
+const (
+	XFrameDeny       XFrameOption = "DENY"
+	XFrameSameOrigin              = "SAMEORIGIN"
+	XFrameAllowFrom               = "ALLOW-FROM"
+
+	CSPSrcSelf CSPSource = iota
+	CSPSrcParent
+	CSPSrcParentSubdomains
+	CSPSrcAny
+)
+
+func Secure(conf *SecureConfig) echo.MiddlewareFunc {
```

Comment:
I don't see where this middleware is used for the main domain (`https://alice.example.com`)